### DISCUSSION TITLE
Contrast and Shadows nob not recognize preset value

### DISCUSSION THREAD

**Discussion by kindian123** [rsjaffe/MIDI2LR#697] [Problems]
Hi, after updating to the latest release I have this issue. Currently using updated Lightroom with Behringer X Touch Mini and latest MIDI2LR. I have a preset with +30 contrast, +30 shadows, using Behringer top nobs before with no issues. Now I have to drag them down to reach something above +30, I guess it starts Contrast with +54 or something when it's actually at +30, doesn't always happen with Shadows. This issue always happened to me with the vertical slider never with the top nobs. Anyone with this issue?

Update: tested the Quick Develop Contrast and Quick Develop Shadows and it works fine, too bad because I have only -5/+5 steps not -1/1 like in Basic Tone

**Comment by adrianpytlikadrian** [comment]
I have a problem with the contrast and shadow knob too. The values are "stuck" on some number and to change the value you have to go back to the given value to make it work. 
<img width="281" alt="Screenshot 2022-11-18 at 22 02 43" src="https://user-images.githubusercontent.com/118570064/202802074-0601af19-a8e8-4f0e-821e-72455681901b.png">

**Reply by ETFocused** [reply]
I have the same problem ... I assume I can just use the previous version to overcome this problem

**Comment by BunnyBread** [comment]
I have this same problem. I'm a total amateur with LUA, but it looks like the changes to the ClientUtilities.lua FullRefresh function in 5.4.0.0 might have something to do with it. Maybe there's a better way, but changing the AltParameter in Database.lua for the contrast and shadow controls to 'Direct' seems to have fixed it for me. I have no idea why only those two controls aren't updating properly.

**Comment by alvinalystra** [comment]
I encountered a similar issue with my X Touch mini. To address it, I made the following change to the 'ClientUtilities.lua' file: In the 'FullRefresh' function, I modified the line 'lrvalue = (photoval[param] or 0) + (photoval[altparam] or 0)' to 'lrvalue = (photoval[altparam] or 0)'. This adjustment appears to have resolved the problem for me.
